board/imgtec/pistachio_bub/otp.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2016 Imagination Technologies
+ * Author: Francois Berder <francois.berder@imgtec.com>
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#include <common.h>
+#include <winbond-otp.h>
+#include "otp.h"
+
+int check_otp_version(loff_t offset)
+{
+	u_char version;
+
+	if (read_otp_data(offset, sizeof(version), (char *)&version)) {
+		printf("WARNING: Could not read register version from OTP.\n");
+		return -1;
+	}
+
+	if (version != 1) {
```

Comment:
I am not sure this arch will work from a backwards compatibility point of view. Imagine we had to release a version 2 OTP layout that was a superset of version 1. The code change should be as simple as here is the extra stuff to read if you see version 2. i.e. it should hit the version 1 code already here (not duplicate it) and then carry on to any further versions.

We would never change layouts already released meaning you can always assume a new version is a superset. So based on this how about we drop this file and just wrap the read_otp calls in conditions of greater than version N (in this case 1)?